server/db-middleware/general-api.js

```diff
@@ -211,4 +211,53 @@ apiModule.getStudentsByTeacherFlat = function (teacherId) {
     .then(() => Object.keys(allKeys));
 };
 
+apiModule.getPersonsCategoried = function (category, skip = 0, top = 10, filterConfig) {
+  if (category !== 'student' && category !== 'teacher') {
+    return new Promise((resolve, reject) => {
+      reject();
+    });
+  }
+  const resUsers = {};
+  let configString = '';
+
+  filterConfig.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, '').split(' ').forEach((el, i) => {
+    if (el !== '') {
+      if (configString === '') {
+        configString += el;
+      } else {
+        configString += '|' + el;
+      }
+    }
+  });
+
+  return User.find({ role: category.toUpperCase() })
+    .find({
+      $or: [{ name: { $regex: configString, $options: 'i' } },
+        { surname: { $regex: configString, $options: 'i' } },
+        { 'account.university': { $regex: configString, $options: 'i' } }],
+    })
+    .select('-__v -role -email -primarySkill -hash -salt')
```

Comment:
Model.find() accepts projection as an argument, so instead of using select, you can write it straight in find()